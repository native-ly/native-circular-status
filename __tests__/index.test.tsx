import React from 'react'
import { render } from '@testing-library/react-native'

import NativeCircularStatus from '../src'

describe('NativeCircularStatus', () => {
  it('should render component', () => {
    const { toJSON } = render(<NativeCircularStatus progress={0} />)

    expect(toJSON()).toMatchSnapshot()
  })

  it.todo('normal thinking')
  it.todo('normal disabled')
  it.todo('normal uncontrolled')
  it.todo('normal renderContent')
  it.todo('normal custom iconPause, iconPlay')
  it.todo('compact')
  it.todo('compact thinking')
  it.todo('compact disabled')
  it.todo('compact uncontrolled')
})
