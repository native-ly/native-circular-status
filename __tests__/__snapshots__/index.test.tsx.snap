// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NativeCircularStatus should render component 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 28,
      "opacity": 1,
      "position": "relative",
      "width": 28,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": "100%",
        "justifyContent": "center",
        "position": "absolute",
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "borderColor": "#efefef",
        "borderRadius": 14,
        "borderWidth": 2,
        "height": "100%",
        "position": "absolute",
        "width": "100%",
      }
    }
  />
  <View
    indeterminateAnimationDuration={1000}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <RNSVGSvgView
      bbHeight={28}
      bbWidth={28}
      focusable={false}
      height={28}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 28,
            "width": 28,
          },
        ]
      }
      width={28}
    >
      <RNSVGGroup>
        <RNSVGPath
          d="M14 1
          A13 13 0 0 1 14 1"
          propList={
            Array [
              "stroke",
              "strokeWidth",
              "strokeLinecap",
            ]
          }
          stroke={4294650963}
          strokeLinecap={0}
          strokeWidth={2}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;
